src/shared/components/palette/judicial-user/write-judicial-user-field.component.ts

```diff
@@ -0,0 +1,73 @@
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+import { catchError, debounceTime, filter, switchMap, tap } from 'rxjs/operators';
+import { JudicialUserModel } from '../../../domain/jurisdiction';
+import { JurisdictionService } from '../../../services';
+import { AbstractFieldWriteComponent } from '../base-field';
+
+@Component({
+  selector: 'ccd-write-judicial-user-field',
+  styleUrls: ['./write-judicial-user-field.component.scss'],
+  templateUrl: './write-judicial-user-field.component.html'
+})
+export class WriteJudicialUserFieldComponent extends AbstractFieldWriteComponent implements OnInit, OnDestroy {
+
+  private readonly JURISDICTION_ID = 'jid';
+  private readonly MINIMUM_SEARCH_CHARACTERS = 2;
+  private readonly IDAM_ID = 'idamId';
+  private readonly PERSONAL_CODE = 'personalCode';
+
+  public judicialUserFormGroup: FormGroup;
+  public idamIdFormControl: FormControl;
+  public personalCodeFormControl: FormControl;
+  public jurisdictionId: string;
+  public filteredJudicialUsers: JudicialUserModel[] = [];
+  public showAutocomplete = false;
+  public sub: Subscription;
```

Comment:
@johnbenjamin-hmcts - same as similar comment above

Reply:
@Daniel-Lam - I kept it public mainly to write tests. I also looked in MC and in many places subscription is defined as public and then unit tests were written for it.